Assets/Mirror/Authenticators/BasicAuthenticator.cs

```diff
@@ -35,7 +35,7 @@ public override void OnStartServer()
         public override void OnStartClient()
         {
             // register a handler for the authentication response we expect from server
-            NetworkClient.RegisterHandler<AuthResponseMessage>(OnAuthResponseMessage, false);
+            NetworkManager.singleton.client.RegisterHandler<AuthResponseMessage>(OnAuthResponseMessage, false);
```

Comment:
what about the NetworkManager.singleton calls everywhere?
right now, NetworkClient is static. which imho is way better than the singleton usage here.